app/controllers/activities_controller.rb

```diff
@@ -30,4 +30,28 @@ def poll
       format.js
     end
   end
+
+  private
+
+  def filtering_params
+    params.permit(:user_id, :trackable_type, :since, :before)
+  end
+
+  def filter_activities(activities)
+    filtering_params.each do |key, value|
+      next if value.blank?
+
+      if key == 'since'
+        value = DateTime.parse(value).beginning_of_day rescue ArgumentError
+        activities = activities.since value
+      elsif key == 'before'
+        value = DateTime.parse(value).end_of_day rescue ArgumentError
+        activities = activities.before value
+      elsif ['user_id', 'trackable_type'].include? key
+        activities = activities.where(key.to_sym => value)
```

Comment:
I don't think we need to symbolize

```suggestion
        activities = activities.where(key => value)
```